src/libmongoc/doc/mongoc_auto_encryption_opts_set_extra.rst

```diff
@@ -25,6 +25,34 @@ Parameters
 * ``mongocryptdBypassSpawn`` set to true to prevent the driver from spawning the mongocryptd process (default behavior is to spawn).
 * ``mongocryptdSpawnPath`` set to a path (with trailing slash) to search for mongocryptd (defaults to empty string and uses default system paths).
 * ``mongocryptdSpawnArgs`` set to an array of string arguments to pass to ``mongocryptd`` when spawning (defaults to ``[ "--idleShutdownTimeoutSecs=60" ]``).
+* ``csflePath`` - Set a filepath string referring to a ``csfle`` dynamic library
+  file. Default is undefined.
+
+  * If not specified (the default), ``libmongocrypt`` will attempt to load
+    ``csfle`` using the host system's default dynamic-library-search system.
+  * If used, should be the path to the ``csfle`` dynamic library file itself,
+    not just the directory that contains it.
+  * If the given path is a relative path and the first path component is
+    ``$ORIGIN``, the ``$ORIGIN`` component will be replaced with the absolute
+    path to the directory containing the ``libmongocrypt`` library in use by the
+    application.
+
+    .. note:: No other ``RPATH``/``RUNPATH``-style substitutions are availbale.
+
+  * If the given path is a relative path, the path will be resolves relative to
+    the working directory of the operating system process.
+  * If this option is specified and ``libmongocrypt`` fails to load ``csfle``
+    from that filepath, ``libmongocrypt`` will fail to initialize with a
+    hard-error, and will not attempt to search for ``csfle`` in any other
+    locations.
+
+* ``csfleRequired`` - If set to ``true``, and ``libmongocrypt`` fails to load a
+  ``csfle`` dynamic library, generate a hard-error and do not attempt to spawn
```

Comment:
Does a "hard-error" constitute something more than libmongocrypt initialization failure?

Reply:
The "hard-error" means that initialization fails rather than error-and-continue (as with the case of searching). This should be made more clear.